code/modules/projectiles/ammunition/_firing.dm

```diff
@@ -16,10 +16,8 @@
 		AddComponent(/datum/component/pellet_cloud, projectile_type, pellets)
 		SEND_SIGNAL(src, COMSIG_PELLET_CLOUD_INIT, target, user, fired_from, randomspread, spread, zone_override, params, distro)
 
-	if(click_cooldown_override)
-		user.changeNext_move(click_cooldown_override)
-	else
-		user.changeNext_move(CLICK_CD_RANGE)
+	user.changeNext_move((click_cooldown_override || CLICK_CD_RANGE) * (HAS_TRAIT(user, TRAIT_DOUBLE_TAP) ? 0.5 : 1))
```

Comment:
calculate the scalar first, then do the if thing. this is gonna fuck some people over